plugins/Dashboard/javascripts/dashboard.js

```diff
@@ -29,11 +29,12 @@ function createDashboard() {
             function (response) {
                 var id = response.value;
                 angular.element(document).injector().invoke(function ($location, reportingMenuModel, dashboardsModel) {
-                    dashboardsModel.reloadAllDashboards().then(function () {
-
-                        $('#dashboardWidgetsArea').dashboard('loadDashboard', id);
-                        $('#dashboardWidgetsArea').dashboard('rebuildMenu');
-                    });
+                  Promise.all([
+                    dashboardsModel.reloadAllDashboards(),
+                    reportingMenuModel.reloadMenuItems(),
+                  ]).then(function () {
+                    $('#dashboardWidgetsArea').dashboard('loadDashboard', id);
+                  });
```

Comment:
This actually breaks the dashboad menu in widgetize mode. `$('#dashboardWidgetsArea').dashboard('rebuildMenu');` still needs to be triggered as this would also update the widgetize menu. See 
https://github.com/matomo-org/matomo/blob/934eb450419edaf4acf2fc21e3b52723f11c40f1/plugins/Dashboard/javascripts/dashboardObject.js#L535-L597